substrate/frame/broker/src/benchmarking.rs

```diff
@@ -191,6 +191,25 @@ mod benches {
 		Ok(())
 	}
 
+	#[benchmark]
+	fn remove_lease() -> Result<(), BenchmarkError> {
+		let task = 1u32;
+		let until = 10u32.into();
+
+		// Assume Leases to be almost filled for worst case
+		setup_leases::<T>(T::MaxLeasedCores::get(), task, until);
+
+		let origin =
+			T::AdminOrigin::try_successful_origin().map_err(|_| BenchmarkError::Weightless)?;
+
+		#[extrinsic_call]
+		_(origin as T::RuntimeOrigin, task);
```

Comment:
the worst case would be removing the last task in the vec. Here it removes the first one.

Though in practice the cost should be negligible as `MaxLeasedCores` isn't so big.

Reply:
Agreed, would be nice to target the worst case, especially here where it's as straightforward as just changing the task id.